server.js

```diff
@@ -1,120 +1,75 @@
-        })
-    } else if (postedHeight < 50 || postedHeight > 250) {
-        res.status(400).json({
-            message: 'The height of this person should be, between 50 and 250!',
-        })
+// Get user bij Id.
+app.get("/users/:Id", (req, res) => {
+    const userFound = myUtilityFunctions.getUserById(req.params.Id);
+    if(userFound) {
+      res.status(200).json({
+        message: 'Successfully retrieved a specific user!',
+        result: userFound,
+      });
     } else {
-        const fs = require('fs');
-        const path = './user.json';
-        if (fs.existsSync(path) != true) {
-            fs.appendFileSync('user.json', JSON.stringify([req.body], null, 2));
-        } else {
-            const fileData = JSON.parse(fs.readFileSync('user.json'))
-            fileData.push(req.body)
-            fs.writeFileSync('user.json', JSON.stringify(fileData, null, 2));
-        }
-        const jsonData = require('./user.json');
-        res.status(201).json({
-            message: 'Successfully created a specific user!',
-            result: jsonData,
-        })
+      res.status(404).json({
+        message: 'Not Found',
+      })
     }
-})
+  });
 
-// get one specific user
-app.get("/users/:Id", (req, res) => {
-    const jsonData = require('./user.json');
-    const requestedObject = req.params.Id;
-    let indexGet = jsonData.findIndex(requestedObject => {
-        return requestedObject.name === `${req.params.Id}`;
-    });
-    if (indexGet > -1) {
-        res.status(200).json({
-            message: 'Successfully retrieved a specific user!',
-            result: jsonData[indexGet]
+// Validate and add users to the user.json, if the users file does not exist, creates the file with append.
+app.post("/users", (req, res) => {
+    const validation = (myUtilityFunctions.validationUserPost(req.body) == true);
+    if (validation) {
+        res.status(201).json({
+            message: `The user ${req.body.name} is succesfully created!`
         });
+        return myUtilityFunctions.addUserPost(req.body);
     } else {
-        res.status(404).json({
-            message: 'Not Found',
+        res.status(401).json ({
+            message: 'The user cannot be validated!'
         })
+        return validation;
     }
 });
 
-//now i can update any parameter of users, ot a total user if i change a little bit at line 58
+// The first if add a user and creates the file if the user.json file does not exist yet.
 app.put("/users/:Id", (req, res) => {
-    const jsonData = require('./user.json');
-    const updatedObject = req.params.Id;
-    let indexPut = jsonData.findIndex(updatedObject => {
-        return updatedObject.name === `${req.params.Id}`;
-    });
-    if (indexPut > -1) {
-        jsonData[indexPut] = req.body;
-        res.status(200).json
+    if (myUtilityFunctions.putUserbyId(req.body, req.params.Id) == true) {
+        console.log(myUtilityFunctions.putUserbyId(req.body, req.params.Id));
```

Comment:
I just forgot to remove this console.log(), it was only because of test purposes, it does show only true indeed.
To store this to a variable is making more code, and for me the readability decreases, or do you see this as an improvement in readability?

Reply:
For the readability increases :)

Normally when you have such a operation that is "important" in the sense that it changes state, write things to file, calls other systems etc, you make it call in a single line and stores the variable for later evaluation, because normally you will want to use this variable multiple times also (for logging, checking the results in more than one way, returning part of its content etc).

Less code isn't always less readability ;)

Reply:
I understand now why! I changed some functions now I see what you mean by better readability:)